wire-runtime/src/commonMain/kotlin/com/squareup/wire/ProtoAdapter.kt

```diff
@@ -356,12 +367,12 @@ internal inline fun <E> ProtoAdapter<E>.commonEncodeByteString(value: E): ByteSt
 
 @Suppress("NOTHING_TO_INLINE")
 internal inline fun <E> ProtoAdapter<E>.commonDecode(bytes: ByteArray): E {
-  return decode(Buffer().write(bytes))
+  return decode(ByteArrayProtoReader(bytes))
 }
 
 @Suppress("NOTHING_TO_INLINE")
 internal inline fun <E> ProtoAdapter<E>.commonDecode(bytes: ByteString): E {
-  return decode(Buffer().write(bytes))
+  return decode(ByteArrayProtoReader(bytes))
```

Comment:
I gotta revert this and the one below before merging. Otherwise we call a generated function that might not exist in already-compiled libraries.